R/read_csv.R

```diff
@@ -204,8 +204,14 @@ read_sample_csv <- function(output_files) {
       id <- csv_file_info$id
     }
     # read sampling data
-    draws <- utils::read.csv(output_file, header = TRUE, comment.char = "#")
-    if (nrow(draws) > 0) {
+    suppressWarnings(      
+      draws <- vroom::vroom(output_file, comment = "# ", delim = ',', trim_ws = TRUE, col_types = c(lp__ = "d"), num_threads = cores)
+    )
+    if(ncol(draws) == 0) {
+      stop("The supplied csv file does not contain any sampling data!")
+    }
+    draws <- draws[!is.na(draws$lp__),]
```

Comment:
This is needed because what vroom reads in is the following:
```
     lp__ accept_stat__ stepsize__ treedepth__ n_leapfrog__ divergent__ energy__   theta
    <dbl>         <dbl>      <dbl>       <dbl>        <dbl>       <dbl>    <dbl>   <dbl>
 1  NA           NA         NA              NA           NA          NA    NA    NA     
 2  NA           NA         NA              NA           NA          NA    NA    NA     
 3  NA           NA         NA              NA           NA          NA    NA    NA     
 4  NA           NA         NA              NA           NA          NA    NA    NA     
 5  -6.87         0.955      0.931           1            3           0     7.08  0.192 
 6  -7.96         0.729      0.931           2            3           0     9.23  0.0950
 7  -7.85         1          0.931           1            1           0     8.07  0.100 
 8 -11.1          0.746      0.931           1            3           0    12.0   0.0272
 9  -8.30         1          0.931           2            3           0    11.1   0.0811
10  -6.84         0.967      0.931           1            3           0     8.53  0.199 
# … with 999 more rows
# A tibble: 1,009 x 8  
```
Those above are there because of our csv format. I think we could get away with always skipping first 4 rows, but this is safer.